### PR TITLE
RES-1452: Add checkpoint backward compatibilty test

### PR DESCRIPTION
**Pull request by lscheinkman**

### CODE REVIEW

File: tests/unit/frameworks/pytorch/imagenet_util_test.py [hunk 1]
```diff
@@ -19,26 +19,62 @@
...
 from nupic.research.frameworks.pytorch.models.resnets import resnet50
 
+CURRENT_DIR = Path(__file__).parent
+
+
+def _create_test_model():
+    """
+    Create standard resnet50 model to be used in tests.
+    """
+    model = create_model(model_class=resnet50, model_args={},
```
[the remark this answers is not in the document]
Comment (lscheinkman) [reply tests/unit/frameworks/pytorch/imagenet_util_test.py]: Updated the tests to include sparse models. 
Keep in mind this PR stores the checkpoints using [git-lfs](https://git-lfs.github.com/) and you may need to install the git-lfs on your computer to be able to download the checkpoints

File: tests/unit/frameworks/pytorch/imagenet_util_test.py [hunk 2]
```diff
@@ -19,26 +19,80 @@
...
+    serialize_state_dict,
+    set_random_seed,
+)
 from nupic.research.frameworks.pytorch.models.resnets import resnet50
+from nupic.torch.modules import rezero_weights
+
+CHECKPOINTS_DIR = Path(__file__).parent / "checkpoints"
+
+TEST_MODEL_ARGS = dict(config=dict(
+    activation_params_func=lambda *_: dict(percent_on=0.5, local=True),
```
Comment (subutai) [inline tests/unit/frameworks/pytorch/imagenet_util_test.py:39]: Don't we have to set 'defaults_sparse=True'?

Comment (lscheinkman) [reply tests/unit/frameworks/pytorch/imagenet_util_test.py:39]: I am overriding the default sparse to create smaller checkpoints, we just care about the network structure for checkpoints. The default sparse creates large checkpoints (~100M)

Comment (subutai) [reply tests/unit/frameworks/pytorch/imagenet_util_test.py:39]: 👍

**Pull request merged**